frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -151,118 +129,137 @@ class PopupCreateFile extends React.Component {
+      xhr.upload.addEventListener('progress', this.uploadInProgress(file), false)
 
-    xhr.setRequestHeader('Accept', 'application/json')
-    xhr.withCredentials = true
+      xhr.open('POST', `${state.config.apiUrl}/workspaces/${state.workspaceId}/files`, true)
 
-    xhr.onreadystatechange = () => {
-      if (xhr.readyState === 4) {
-        const uploadedFileUpdatedList = state.uploadedFileList
-        switch (xhr.status) {
-          case 200:
-            const jsonResult200 = JSON.parse(xhr.responseText)
+      xhr.setRequestHeader('Accept', 'application/json')
+      xhr.withCredentials = true
 
-            filePosted.jsonResult = { ...jsonResult200, code: 200 }
+      xhr.onerror = () => reject(xhr)
 
-            uploadedFileUpdatedList.push(filePosted)
-            this.setState({ uploadedFileList: uploadedFileUpdatedList })
-            break
-          case 400:
-            const jsonResult400 = JSON.parse(xhr.responseText)
-
-            let errorMessage = props.t('Error while creating file')
-            switch (jsonResult400.code) {
-              case 3002: errorMessage = props.t('A content with the same name already exists'); break
-              case 6002: errorMessage = props.t('The file is larger than the maximum file size allowed'); break
-              case 6003: errorMessage = props.t('Error, the shared space exceed its maximum size'); break
-              case 6004: errorMessage = props.t('You have reach your storage limit, you cannot add new files'); break
-            }
-            filePosted.jsonResult = jsonResult400
-            filePosted.errorMessage = errorMessage
-            uploadedFileUpdatedList.push(filePosted)
-            this.setState({ uploadedFileList: uploadedFileUpdatedList })
-            break
-          default:
-            filePosted.jsonResult = { code: 0 }
-            filePosted.errorMessage = props.t('Error while creating file')
-            uploadedFileUpdatedList.push(filePosted)
-            this.setState({ uploadedFileList: uploadedFileUpdatedList })
-            break
-        }
+      xhr.onreadystatechange = () => {
+        if (xhr.readyState === 4) resolve(xhr)
       }
-    }
 
-    xhr.send(formData)
+      xhr.send(formData)
+    })
   }
 
   handleValidate = async () => {
-    const { state } = this
+    const { state, props } = this
 
     this.setState({ uploadStarted: true })
 
-    state.uploadFileList.forEach((file, index) => this.postFile(file, index))
+    Promise.all(state.fileToUploadList.map(async (file) => {
+      const xhr = await this.postFile(file)
+      return this.handleFileUploadEnd(xhr, file)
+    })).catch(() => {
+      this.sendGlobalFlashMessage(props.t('Error while creating file'))
+    }).then(uploadedFiles => {
+      this.handleAllFileUploadEnd(uploadedFiles)
+    })
+  }
+
+  handleFileUploadEnd = (xhr, file) => {
+    const { props } = this
+
+    const filePosted = file
```

Comment:
by doing this, the function isn't immutable since you don't create a new object.
Find a way of creating a new object of file or, if it's not possible, don't use another variable